src/crimson/osd/osd_operation_sequencer.h

```diff
@@ -61,9 +61,13 @@ class OperationRepeatSequencer {
     });
   }
 
-  void finish_op(OpRef& op, const spg_t& pgid) {
+  void finish_op(OpRef& op, const spg_t& pgid, bool interrutped) {
     assert(op->pos);
-    pg_ops.at(pgid).erase(*(op->pos));
+    auto curr_op_pos = *(op->pos);
+    if (interrutped) {
+      curr_op_pos->second.set_value();
```

Comment:
I'm still thinking that maybe we should directly set an exception like "op_abort" here, as this line would allow other ops to go on executing without the support of interruptible future. But, since we are going to merge that op sequncer refactor pr which would do exactly what is mentioned above, I think this line is acceptable:-)